From 73795095b288db1fcd5728981b5179ccd935bbe3 Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Fri, 14 Nov 2025 01:44:10 +0100
Subject: [PATCH 35/36] Fix build after
 1f79bc48382cc204a9cb0eae1d3cca2515af1f3c

After this change the build requires definitions for:

* __truncl, __trunc, __truncf
* __ceill, __ceil, __ceilf

...and fails if not provided:
```
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_fromfpl.os): in function `__fromfpl0':
s_fromfpl.c:(.text+0x72): undefined reference to `__truncl'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_fromfpl.c:(.text+0xa2): undefined reference to `__ceill'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_fromfpxl.os): in function `__fromfpxl0':
s_fromfpxl.c:(.text+0x82): undefined reference to `__truncl'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_fromfpxl.c:(.text+0xb2): undefined reference to `__ceill'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_ufromfpl.os): in function `__ufromfpl0':
s_ufromfpl.c:(.text+0x72): undefined reference to `__truncl'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_ufromfpl.c:(.text+0xa2): undefined reference to `__ceill'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_ufromfpxl.os): in function `__ufromfpxl0':
s_ufromfpxl.c:(.text+0x82): undefined reference to `__truncl'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_ufromfpxl.c:(.text+0xb2): undefined reference to `__ceill'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_fromfp.os): in function `__fromfp0':
s_fromfp.c:(.text+0x62): undefined reference to `__trunc'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_fromfp.c:(.text+0x82): undefined reference to `__ceil'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_fromfpx.os): in function `__fromfpx0':
s_fromfpx.c:(.text+0x72): undefined reference to `__trunc'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_fromfpx.c:(.text+0xa2): undefined reference to `__ceil'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_ufromfp.os): in function `__ufromfp0':
s_ufromfp.c:(.text+0x62): undefined reference to `__trunc'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_ufromfp.c:(.text+0x82): undefined reference to `__ceil'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_ufromfpx.os): in function `__ufromfpx0':
s_ufromfpx.c:(.text+0x72): undefined reference to `__trunc'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_ufromfpx.c:(.text+0xa2): undefined reference to `__ceil'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_fromfpf.os): in function `__fromfpf0':
s_fromfpf.c:(.text+0x62): undefined reference to `__truncf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_fromfpf.c:(.text+0x82): undefined reference to `__ceilf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_fromfpxf.os): in function `__fromfpxf0':
s_fromfpxf.c:(.text+0x72): undefined reference to `__truncf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_fromfpxf.c:(.text+0xa2): undefined reference to `__ceilf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_ufromfpf.os): in function `__ufromfpf0':
s_ufromfpf.c:(.text+0x62): undefined reference to `__truncf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_ufromfpf.c:(.text+0x82): undefined reference to `__ceilf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm_pic.a(s_ufromfpxf.os): in function `__ufromfpxf0':
s_ufromfpxf.c:(.text+0x72): undefined reference to `__truncf'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/16.0.0/../../../../ia64-t2-linux-gnu/bin/ld: s_ufromfpxf.c:(.text+0xa2): undefined reference to `__ceilf'
collect2: error: ld returned 1 exit status
make[2]: *** [../Makerules:542: /usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/objs/math/libm.so] Error 1
make[2]: Leaving directory '/usr/src/t2-src/src.glibc.ia64-toolchain.251113.035941.118328/glibc-2.42.9000-487-g1f79bc4838/math'
```

So let's provide the missing definitions via the "libm_alias_[...]" macros,
i.e. the ones w/o "_other" at the end. That helps.
---
 sysdeps/ia64/fpu/s_ceil.S   | 2 +-
 sysdeps/ia64/fpu/s_ceilf.S  | 2 +-
 sysdeps/ia64/fpu/s_ceill.S  | 2 +-
 sysdeps/ia64/fpu/s_trunc.S  | 2 +-
 sysdeps/ia64/fpu/s_truncf.S | 2 +-
 sysdeps/ia64/fpu/s_truncl.S | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/sysdeps/ia64/fpu/s_ceil.S b/sysdeps/ia64/fpu/s_ceil.S
index 641f291121..1353a55cfc 100644
--- a/sysdeps/ia64/fpu/s_ceil.S
+++ b/sysdeps/ia64/fpu/s_ceil.S
@@ -221,4 +221,4 @@ CEIL_UNORM:
 ;;
 
 GLOBAL_LIBM_END(ceil)
-libm_alias_double_other (ceil, ceil)
+libm_alias_double (ceil, __ceil)
diff --git a/sysdeps/ia64/fpu/s_ceilf.S b/sysdeps/ia64/fpu/s_ceilf.S
index 84a95696bf..56ad8dcb07 100644
--- a/sysdeps/ia64/fpu/s_ceilf.S
+++ b/sysdeps/ia64/fpu/s_ceilf.S
@@ -221,4 +221,4 @@ CEIL_UNORM:
 ;;
 
 GLOBAL_LIBM_END(ceilf)
-libm_alias_float_other (ceil, ceil)
+libm_alias_float (ceil, __ceil)
diff --git a/sysdeps/ia64/fpu/s_ceill.S b/sysdeps/ia64/fpu/s_ceill.S
index 582f7493a8..46532561cd 100644
--- a/sysdeps/ia64/fpu/s_ceill.S
+++ b/sysdeps/ia64/fpu/s_ceill.S
@@ -221,4 +221,4 @@ CEIL_UNORM:
 ;;
 
 GLOBAL_LIBM_END(ceill)
-libm_alias_ldouble_other (ceil, ceil)
+libm_alias_ldouble (ceil, __ceil)
diff --git a/sysdeps/ia64/fpu/s_trunc.S b/sysdeps/ia64/fpu/s_trunc.S
index 81add71f65..1909c4bf11 100644
--- a/sysdeps/ia64/fpu/s_trunc.S
+++ b/sysdeps/ia64/fpu/s_trunc.S
@@ -163,4 +163,4 @@ TRUNC_UNORM:
 ;;
 
 GLOBAL_LIBM_END(trunc)
-libm_alias_double_other (trunc, trunc)
+libm_alias_double (trunc, __trunc)
diff --git a/sysdeps/ia64/fpu/s_truncf.S b/sysdeps/ia64/fpu/s_truncf.S
index 7e084cbd32..e01d8b1c0e 100644
--- a/sysdeps/ia64/fpu/s_truncf.S
+++ b/sysdeps/ia64/fpu/s_truncf.S
@@ -163,4 +163,4 @@ TRUNC_UNORM:
 ;;
 
 GLOBAL_LIBM_END(truncf)
-libm_alias_float_other (trunc, trunc)
+libm_alias_float (trunc, __trunc)
diff --git a/sysdeps/ia64/fpu/s_truncl.S b/sysdeps/ia64/fpu/s_truncl.S
index 6fb44df643..e1d5d8ef27 100644
--- a/sysdeps/ia64/fpu/s_truncl.S
+++ b/sysdeps/ia64/fpu/s_truncl.S
@@ -163,4 +163,4 @@ TRUNC_UNORM:
 ;;
 
 GLOBAL_LIBM_END(truncl)
-libm_alias_ldouble_other (trunc, trunc)
+libm_alias_ldouble (trunc, __trunc)
-- 
2.25.1

